runtime/src/common/key_format.rs

```diff
@@ -70,9 +68,7 @@ pub trait KeyFormat {
         if data[0] != Self::prefix() {
             return None;
         }
-        if data.len() < 1 + Self::size() {
-            panic!("key format: malformed input");
-        }
+        assert!(data.len() > Self::size(), "key format: malformed input");
```

Comment:
I'd somehow find this more understandable with `data.len() >= 1 + Self::size()`

Reply:
we could make it more formal

```rust
const VERSION_PREFIX_SIZE: usize = the_size_of_a_u8;
data.len() >= VERSION_PREFIX_SIZE + Self:size()
```

Reply:
Ah I see, I thought you generally preferred the `1+` version. Yeah I can make this case clearer.